src/olympia/accounts/views.py

```diff
@@ -418,7 +418,7 @@ def get_lookup_field(self, identifier):
     def self_view(self):
         return (
             self.request.user.is_authenticated() and
-            self.get_object() == self.request.user)
+            self.kwargs.get('pk') == self.request.user.pk)
```

Comment:
you can't do this.  `get_object()` does permission checks and `self.kwargs.get('pk')` might not be the `id` - it might be a username instead.